integrations/wagmi-v2/README.md

```diff
@@ -18,67 +18,111 @@ and gas estimations.
 
 ## Usage
 
-First install the package.
+### Installation
+
+First install the package and its dependencies.
 
 ```
 npm install @oasisprotocol/sapphire-wagmi-v2 wagmi@2.x viem@2.x
 ```
 
-Next, in your Wagmi config definition, setup Sapphire wrapping for the injected
-provide using `injectedWithSapphire()` and then define the transports using the
-`sapphireHttpTransport()` function.
+### [EIP-6963] Multi Injected Provider Discovery
+
+#### Single chain - Sapphire
+
+When your application connects exclusively to Sapphire networks, the most 
+straightforward approach is to set `sapphireConfig.replaceProviders = true`. 
+This configuration ensures that all EIP-6963 providers are automatically wrapped 
+with end-to-end encryption upon registration.
 
 ```typescript
-import { createConfig } from "wagmi";
+import { http } from "wagmi";
 import { sapphire, sapphireTestnet } from "wagmi/chains";
 import {
-	injectedWithSapphire,
-	sapphireHttpTransport,
+  sapphireLocalnet,
+  createSapphireConfig,
 } from "@oasisprotocol/sapphire-wagmi-v2";
 
-export const config = createConfig({
-	multiInjectedProviderDiscovery: false,
-	chains: [sapphire, sapphireTestnet],
-	connectors: [injectedWithSapphire()],
-	transports: {
-		[sapphire.id]: sapphireHttpTransport(),
-		[sapphireTestnet.id]: sapphireHttpTransport()
-	},
+export const wagmiConfig = createSapphireConfig({
+  sapphireConfig: {
+    replaceProviders: true,
+  },
+  chains: [sapphire, sapphireTestnet, sapphireLocalnet],
+  transports: {
+    [sapphire.id]: http(),
+    [sapphireTestnet.id]: http(),
+    [sapphireLocalnet.id]: http(),
+  },
 });
 ```
 
-Please note that while [EIP-6963] (Multi Injected Provider Discovery) is
-supported by Wagmi it is only possible to wrap the default injected [EIP-1193]
-compatible provider. For this reason you must disable MIPD support in the
-Wagmi configuration as additional discovered providers will not be Sapphire
-wrapped.
+#### Multichain
+
+For multichain applications, you need to carefully manage which providers use 
+end-to-end encryption. It's essential to avoid using encrypted connectors for 
+non-Sapphire chains, and conversely, to ensure Sapphire chains use the 
+appropriate encrypted connectors.
```

Comment:
```suggestion
For multichain applications, you must manage which providers use 
end-to-end encryption. It's essential to avoid using encrypted connectors for 
non-Sapphire chains, and conversely, to ensure Sapphire chains use the 
appropriate encrypted connectors.
```